src/components/lib/sl/sl.c

```diff
@@ -670,10 +673,22 @@ sl_sched_loop(void)
 			}
 
 			sl_cs_exit();
-		} while (pending);
+		} while (pending > 0);
 
 		if (sl_cs_enter_sched()) continue;
 		/* If switch returns an inconsistency, we retry anyway */
 		sl_cs_exit_schedule_nospin();
 	}
 }
+
```

Comment:
I can't figure out the difference, so this really needsa  comment

Reply:
What line/change is this comment referring to?